pkg/kubelet/dockertools/manager.go

```diff
@@ -261,23 +271,53 @@ func (self *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		}
 		dockerContainerName := dockerName.ContainerName
 		c, found := expectedContainers[dockerContainerName]
-		terminationMessagePath := ""
 		if !found {
-			// TODO(dchen1107): should figure out why not continue here
-			// continue
-		} else {
-			terminationMessagePath = c.TerminationMessagePath
+			continue
 		}
-		// We assume docker return us a list of containers in time order
+		terminationMessagePath := c.TerminationMessagePath
+		if containerDone.Has(dockerContainerName) {
+			continue
+		}
+		// We assume docker returns a list of containers sorted reversedly by
+		// time.
 		if containerStatus, found := statuses[dockerContainerName]; found {
-			// Populate last termination state
+			inspected := false
+			var terminationState *api.ContainerState = nil
 			if containerStatus.LastTerminationState.Termination == nil {
 				result := self.inspectContainer(value.ID, dockerContainerName, terminationMessagePath)
```

Comment:
Yes, conatinerDone should've taken care of this. Consolidated the code a bit and removed the duplicated inspection.